### ISSUE TITLE
Filter Results issues when search text containing special chars

### ISSUE THREAD

**Issue by ValentinaPC** [mozilla/addons-server#3248]
Steps to reproduce:
1. Make a search after "+fox" in AMO-stage
2. Observe that filter menu option is "All Add-ons" (are results)
3. Click on "Extensions" from filter menu 
4. Click back the "All Add-ons" option

Expected results:
After step 2 results are listed. After step 4 same results as before are listed.

Actual results:
After step 2 results are listed. After step 4 no results are listed.

Notes/Issues:
Not reproducing if we don't use special chars at search. 
Reproducing using other special chars like ";", "^".
Verified on FF48(Win 7 x64). Issue is reproducing on AMO-stage, dev and prod.
Video for this issue: 
![filter](https://cloud.githubusercontent.com/assets/15685960/17554303/e855fda6-5f12-11e6-8bba-7286544b5b69.gif)

**Comment by EnTeQuAk** [comment]
This should have been fixed via https://github.com/mozilla/addons-server/pull/7303 - at least it contains a few more test-cases that use special characters and don't fall over. Let me know if things are still broken, but I think it might make sense to file new issues about new findings if possible.

**Comment by ioanarusiczki** [comment]
I verified this scenario in classic view on AMO dev, FF58 (Win10) and I noticed that after a couple of clicks on the categories displayed at one point "No results found" is displayed .  All the previous categories which displayed results will return now "No results found" too.  I tried with ^fire, +fire, also without special characters using "Show my Password", could not reproduce it with another extension name though.

![3248search](https://user-images.githubusercontent.com/33448286/35570126-ebf72626-05d6-11e8-98b5-23e912699d11.gif)

![3248search2](https://user-images.githubusercontent.com/33448286/35570127-ec1e19f2-05d6-11e8-83a7-a27f0dac20da.gif)

Using this scenario I made some testing on the new front end and it seems to be alright there.
